libusb/os/darwin_usb.c

```diff
@@ -1802,10 +1802,18 @@ static int darwin_reenumerate_device (struct libusb_device_handle *dev_handle, b
   usbi_dbg (ctx, "darwin/reenumerate_device: waiting for re-enumeration to complete...");
 
   time = 0;
+  struct timespec start = { .tv_sec = 0, .tv_nsec = 0};
+  clock_gettime(CLOCK_MONOTONIC, &start);
+
   while (dpriv->in_reenumerate) {
     struct timespec delay = {.tv_sec = 0, .tv_nsec = 1000};
     nanosleep (&delay, NULL);
-    if (time++ >= DARWIN_REENUMERATE_TIMEOUT_US) {
+
+    struct timespec now = { .tv_sec = 0, .tv_nsec = 0 };
```

Comment:
The initialization here and above is unnecessary. ```clock_gettime()``` should not fail in this case so the structure will always be populated.